docs/extensibility/payment-method-integration.md

```diff
@@ -111,35 +113,35 @@ A big part of the payment method integration is the interface that is exposed fo
+-   `components`: The properties on this object are exposed components that can be implemented by your payment method for various common interface elements used by payment methods. Currently the available components on this property are: `ValidationInputError` (a container for holding validation errors which typically you'll include after any inputs), [`PaymentMethodLabel`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e089ae17043fa525e8397d605f0f470959f2ae95/assets/js/payment-method-extensions/payment-methods/paypal/index.js#L37-L40) (use this component for the payment method label, including an optional icon)
 -   `setExpressPaymentError`: This function receives a string and allows express payment methods to set an error notice for the express payment area on demand. This can be necessary because some express payment method processing might happen outside of checkout events.
 
 ## Server Side Integration
 
-> **Note:** This portion of the doc is still in development and there could be changes to the API here.
-
 ### Processing Payment
 
-Currently, the checkout block has legacy handling for payment processing (by converting incoming `payment_data` provided by the client side payment method to `$_POST` and calling the payment gateway's `process_payment` function ). So unless your payment method hooks into the core checkout `process_checkout` function in any way, this will take care of processing your payment for you.
+Currently, the checkout block has legacy handling for payment processing (by converting incoming `payment_data` provided by the client side payment method to `$_POST` and calling the payment gateway's `process_payment` function ). So unless your payment method hooks into the core checkout `process_checkout` function in any way, this will take care of processing your payment for you server side if you already have a WooCommerce Payment method extension integrated with the existing shortcode checkout flow.
 
 ### Registering Assets
 
-Since implementing the correct loading of your client side asset registration is tricky with the variables around the block loading process, there is an API for ensuring you can register any assets and data to pass along to your client side payment method from the server.
+Since implementing the correct loading of your client side asset registration is tricky with some dependencies on how related assets load in the request, there is an API for ensuring you can register any assets and data to pass along to your client side payment method from the server.
```

Comment:
I had a bit of trouble understanding the wording here.  Are you saying that the abstract class (and interface) below are available to help payment method developers make sure that their client-side scripts get registered/enqueued/served correctly alongside related scripts for WC core, Blocks, etc.?

Reply:
Yes, basically what happens is the payment method registers their handle and the payment method registry surfaces these handles to be registered as dependencies on the actual block assets. The reason this is needed because while the `WP_Scripts` dependency handling works well for indicating what scripts need to load _before_ my JS file, there's no built in way for me to indicate that my script must load _before_ another registered script in another extension. 

I see how I can reword this a bit so I'm going to give a go.